gap/PolygonalComplexes/properties.gi

```diff
@@ -218,6 +218,150 @@ InstallMethod( FaceCounter, "for a twisted polygonal complex",
         return Collected( Compacted( faceDegrees ) );
     end
 );
+
+InstallMethod( ButterflyCounter, "for a simplicial surface",
+    [IsSimplicialSurface],
+    function(surface)
+	local e,temp1,temp2,counter,g,VerticesOfOrthogonalEdge;
+
+        VerticesOfOrthogonalEdge:=function(S,e)
+            local temp;
+            temp:=FacesOfEdge(S,e);
+            temp:=Union(VerticesOfFace(S,temp[1]),VerticesOfFace(S,temp[2]));
+            return Difference(temp, VerticesOfEdge(S,e));
+	end;
+	counter:=[];
+	for e in InnerEdges(surface) do 
+		temp1:=VerticesOfEdge(surface,e);
+		temp1:=SortedList(List(temp1,g->FaceDegreeOfVertex(surface,g)));
+		if not 2 in temp1 then 
+			temp2:=VerticesOfOrthogonalEdge(surface,e);
+			temp2:=SortedList(List(temp2,g->FaceDegreeOfVertex(surface,g)));
+		else
+			temp2:=FacesOfEdge(surface,e)[1];
+			temp2:=VerticesOfFace(surface,temp2);
+			temp2:=Filtered(temp2,g-> not g in VerticesOfEdge(surface,g));
+			temp2:=SortedList(List(temp2,g->FaceDegreeOfVertex(surface,g)));
+		fi;			
+		Add(counter,[temp1,temp2]);
+	od;
+	return Set(Collected(counter));
+end
+);
+
+
+InstallMethod( UmbrellaCounter, "for a simplicial surface",
+    [IsSimplicialSurface],
+    function(surface)
+	local n,counter,v,temp,orb,tup,G,perm,OrbitOnList,tempcounter,temp1,rightumb,bub;
+	tempcounter:=[];
+	rightumb:=function(L)
+		local m,temp,i,t;
+		temp:=orb;		
+		n:=Length(temp[1]);
+		i:=2;
+		m:=Minimum(temp[1]);
+		temp:=Filtered(temp,g->g[1]=m);
+		while temp <>[] and i<=n do 
+			m:=Minimum(List(temp,g->g[i]));
+			temp:=Filtered(temp,g->g[i]=m);
+			i:=i+1;
+			if Length(temp)=1 then
+				return temp[1];
+			fi; 
+				
+		od;
+		return temp[1];
+	end;
+
+	OrbitOnList:=function(G,L)
+		local g,orb,temp,temp1,i,l,n;
+		orb:=[];
+		n:=Length(L);
+		l:=[1..n];
+		temp1:=[];
+		for g in G do
+			temp:=[]; 
+			for i in [1..n] do
+				temp[i]:=L[i^g];
+			od;
+			Add(temp1,temp);
+		od;
+		return temp1;
+	end;
+	bub:=function(list)
+		local g,tempm,tempL,m;
+		tempL:=[];
+		tempm:=List(list,g->Length(g));
+		for m in Set(tempm) do
+			temp:=Filtered(list,g->Length(g)=m);
+			Append(tempL,temp);
+		od; 
+		return tempL;
+	end;
+	for v in Vertices(surface) do
+		temp:=UmbrellaPathOfVertex(surface,v);
+		temp:=EdgesAsList(temp);
```

Comment:
If you here use VerticesAsList(temp) you will not use the two following lines are not needed.

Reply:
UmbrellaPathOfVertex returns an Edge-Face-Path, so Running VerticesAsList would result in an error

Reply:
Oh, sorry! That's right.